bcda/models/postgres/repository.go

```diff
@@ -509,6 +509,22 @@ func (r *Repository) GetJobKeys(ctx context.Context, jobID uint) ([]*models.JobK
 	return keys, nil
 }
 
+func (r *Repository) GetJobKey(ctx context.Context, jobID uint, fileName string) (*models.JobKey, error) {
+	sb := sqlFlavor.NewSelectBuilder().Select("id", "file_name", "resource_type").From("job_keys")
+	sb.Where(sb.And(sb.Equal("job_id", jobID), sb.Equal("file_name", fileName)))
+
+	query, args := sb.Build()
+	row := r.QueryRowContext(ctx, query, args...)
+
+	jk := models.JobKey{JobID: jobID}
+
+	if err := row.Scan(&jk.ID, &jk.FileName, &jk.ResourceType); err != nil {
```

Comment:
If the select statement returns 0 rows, will Scan return an error or will it just fill null values into the struct?

Reply:
Using a filename that wasn't in the DB, it seemed to generate an error.